### PR TITLE
M3-5679: Querify database credentials reset

### PR DESCRIPTION
**Pull request by codyfinn**
## Description

Creates a query for resetting a database's credentials. The query invalidates and removes the cached response from the get credentials query.

## How to test

1. Navigate to a database cluster's details page
2. Click the show button for the root database password
3. Take note of the current root database password
4. Navigate to the settings tab
5. Click the Reset Root Password button
6. Click the confirm button in the confirmation modal
7. Navigate back to the summary tab
8. Click the show button for the root database password
9. Ensure that the new root password is different from the one from step 3.

### CODE REVIEW

**Comment by bnussman** [comment]
@codyfinn Try [this](https://github.com/codyfinn/manager/compare/M3-5679-query-for-reseting-database-creds...bnussman:reset-creds-testing) out and let me know what you think.

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettingsResetPasswordDialog.tsx [hunk 1]
```diff
@@ -45,23 +46,23 @@ const renderActions = (
...
   const { open, onClose, databaseEngine, databaseID } = props;
 
-  const [isLoading, setIsLoading] = React.useState(false);
-  const [error, setError] = React.useState('');
+  const [error, setError] = React.useState<string | undefined>();
+
+  const { mutateAsync, isLoading } = useDatabaseCredentialsMutation(
+    databaseEngine,
+    databaseID
+  );
```
Comment (bnussman) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettingsResetPasswordDialog.tsx:54]:
```suggestion
  const { mutateAsync, isLoading, error } = useDatabaseCredentialsMutation(
    databaseEngine,
    databaseID
  );
```

Source the error from here and remove the `error` state.

**Review by bnussman — CHANGES_REQUESTED** [review]

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx [hunk 2]
```diff
@@ -106,8 +106,7 @@ export const DatabaseSummaryConnectionDetails: React.FC<Props> = (props) => {
     refetch: getDatabaseCredentials,
   } = useDatabaseCredentialsQuery(database.engine, database.id);
 
-  const password =
-    showCredentials && credentials ? credentials?.password : '••••••••';
```
Comment (bnussman) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx:110]: May want to revert back to this, I think this got changed unintentionally.

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx [hunk 3]
```diff
@@ -106,8 +106,7 @@ export const DatabaseSummaryConnectionDetails: React.FC<Props> = (props) => {
...
+  const password = showCredentials ? credentials?.password : '••••••••';
```
Comment (bnussman) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx:109]:
```suggestion
  const password =
    showCredentials && credentials ? credentials?.password : '••••••••';
```

**Review by bnussman — APPROVED** [review]
Revert the noted item and this should be good to go!

**Review by HanaXu — APPROVED** [review]

**Review by bnussman — APPROVED** [review]

**Pull request merged**